tsfc/driver.py

```diff
@@ -365,6 +365,16 @@ def lower_integral_type(fiat_cell, integral_type):
 
 def pick_mode(mode):
     "Return one of the specialized optimisation modules from a mode string."
+    try:
+        from firedrake_citations import Citations
+        cites = {"vanilla": ("Homolya2017", ),
+                 "coffee": ("Luporini2016", "Homolya2017", ),
+                 "spectral": ("Luporini2016", "Homolya2017", "Homolya2017a"),
+                 "tensor": ("Homolya2017", )}
```

Comment:
I would say [doi:10.1145/1163641.1163644](https://doi.org/10.1145/1163641.1163644) deserves a citation in tensor mode. Though I don't think anyone uses or knows about tensor mode.

Reply:
Agreed, added.